### ISSUE TITLE
Is using BatchNormalization even worth the huge additional computation time?

### ISSUE THREAD

**Issue by madeofwin** [fchollet/keras#3352]
While I observed that BatchNormalization is speeding up the convergence, it also adds A LOT to the training time. When I use BatchNormalization after each of my Conv layers for example, my epoch time rises from ~10 seconds to multiple minutes. Given that I could train dozens of non-BN-epochs for each BN-epoch, I'm sceptical if using BatchNormalization in Keras is worth the additional training time.

Thoughts? Is there any other benefit?

**Comment by EdwardRaff** [comment]
There may be something odd with your setup. I've never had BN add that dramatic an amount of time. Perhaps try playing around with the theano settings.

**Comment by madeofwin** [comment]
Thanks for your input. These are my current theano settings:

```
[global]
floatX = float32
device = gpu

[lib]
cnmem = 0.5

[nvcc]
optimizer_including=cudnn
fastmath=true
```

Do you have any suggestions what I could change that could have an impact on the BatchNormalization speed?

**Comment by nouiz** [comment]
Make sure that Keras use the new Theano cudnn wrapper of batch-norm. There
was other discussion on git about this. I forgot if the change was merge in
keras or not.

This will give compuation and compilation time speed up.

I saw report from different people that in some use case, the old BN didn't
make computation slower, but in other it was doing this. It is in those
case that there was good speed up by using cudnn batch norm.

**Comment by madeofwin** [comment]
Keras is using cudnn's BatchNormalization now in it's new version, huge speedup - problem solved!
